code/go/0chain.net/smartcontract/storagesc/alloation2_test.go

```diff
@@ -322,7 +322,7 @@ func TestFinalizeAllocation(t *testing.T) {
 
 	t.Run(ErrFinalizedTooSoon, func(t *testing.T) {
 		var allocationExpired = allocation
-		allocationExpired.Expiration = now - toSeconds(allocation.ChallengeCompletionTime) + 1
+		allocationExpired.Expiration = now - toSeconds(getMaxChallengeCompletionTime()) + 1
```

Comment:
Replace `getMaxChallengeCompletionTime()` with conf.MaxChallengeCompletionTime

Also [getMaxChallengeCompletionTime() ](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/models.go#L1337) should be removed. This pre existing code is wong
```go
func getMaxChallengeCompletionTime() time.Duration {
	return config.SmartContractConfig.GetDuration(confMaxChallengeCompletionTime)
}
```
We should get the settings from the MPT not the config file. If the value of challenge completion time has changed, then the settings file will be out of date.

Instead, use
```go
conf.MaxChallengeCompletionTime
```
Where `conf` well have been previous got via something like
```go
	if conf, err = getConfig(balances); err != nil {
		return nil, 0, fmt.Errorf("can't get config: %v", err)
	}
```
only get the config from the MPT once per smart contract.

Reply:
Before we finish the code for storing/updating global configs in MPT, we can read it from yaml file atm.